### PR TITLE
Eventhub

### PR DESCRIPTION
**Pull request by aliostad**
Here is is - as discussed.

### CODE REVIEW

File: test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj [hunk 1]
```diff
@@ -63,7 +67,6 @@
     </Reference>
     <Reference Include="Microsoft.ServiceBus, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\packages\WindowsAzure.ServiceBus.3.4.1\lib\net45-full\Microsoft.ServiceBus.dll</HintPath>
-      <Private>True</Private>
```
Comment (sergii-sakharov) [inline test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj]: see prev

File: test/ConveyorBelt.Tooling.Test/Parsing/JsonCovert.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace ConveyorBelt.Tooling.Test.Parsing
+{
+    internal class JsonCovert
```
Comment (sergii-sakharov) [inline test/ConveyorBelt.Tooling.Test/Parsing/JsonCovert.cs]: Is this class necessary?

File: test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj [hunk 3]
```diff
@@ -118,6 +124,8 @@
     <Compile Include="IisLogTests.cs" />
     <Compile Include="IndexNamerTests.cs" />
     <Compile Include="IntegrationTests.cs" />
+    <Compile Include="Parsing\InsightMetricsParserTests.cs" />
+    <Compile Include="Parsing\JsonCovert.cs" />
```
Comment (sergii-sakharov) [inline test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj]: Is this file necessary?

File: test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs [hunk 4]
```diff
@@ -67,7 +67,7 @@ public void ParseCompleteErrorLevel()
             }
         }
 
-        [Fact]
+        [Fact(Skip ="Not used anymore")]
```
Comment (sergii-sakharov) [inline test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs]: why skip vs remove from code?

File: test/ConveyorBelt.Tooling.Test/Parsing/InsightMetricsParserTests.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+			"timeGrain": "PT1M"
+            */
+            try
+            {
+                var parser = new InsightMetricsParser();
+                var records = parser.Parse(stream, null).ToList();
+
+                Assert.Equal(96, records.Count);
+                var r = records[0];
+                var ts = r.Timestamp;
```
Comment (sergii-sakharov) [inline test/ConveyorBelt.Tooling.Test/Parsing/InsightMetricsParserTests.cs]: was probably intended to be used below?

File: test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj (hunk 1 above)
Comment (aliostad) [reply test/ConveyorBelt.Tooling.Test/ConveyorBelt.Tooling.Test.csproj]: It is VS doing it.

File: src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs [hunk 6]
```diff
@@ -29,5 +29,23 @@ public static DateTimeOffset GetEventDateTimeOffset(this DynamicTableEntity enti
             // OK timestamp is the only thing we can look for then
             return entity.Timestamp;
         }
+
+        public static DateTimeOffset GetTimestamp(this DynamicTableEntity entity, DiagnosticsSourceSummary source)
+        {
+
```
[the remark this answers is not in the document]
Comment (aliostad) [reply src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs]: not sure what you mean. It is constant and is not really useful for any other thing

[the remark this answers is not in the document]
Comment (sergii-sakharov) [reply src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs]: What I mean is that currently this function evaluates datetime fieldName for every single document that is pushed to elasticsearch. That makes it a fairly hot method.

[the remark this answers is not in the document]
Comment (aliostad) [reply src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs]: Is there a problem with it? Would you think checking existence of a field in a dictionary (most cases returns false) can slow down?

File: test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs (hunk 4 above)
Comment (aliostad) [reply test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs]: Create an issue to get rid of SiteCore. This has nothing to do with the changes I made.

**Comment by aliostad** [comment]
@sergii-sakharov  Do you wanna compare the performance? Memory should not be a problem, in fact I think we should move to bigger sizes such as A4_v2.

File: test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs (hunk 4 above)
Comment (aliostad) [reply test/ConveyorBelt.Tooling.Test/Parsing/SitecoreLogParserShould.cs]: Created.

File: src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (sergii-sakharov) [reply src/ConveyorBelt.Tooling/Internal/TableEntityExtensions.cs]: To be fair it's definitely nowhere close to becoming perf bottleneck anytime soon. It's worth noting though that DynamicProperties is a ConcurrentDictionary and is not as performant as a normal one (heavy usage can cause lock contention). This concern should be easy to clear with just a run of CB e.g. spotting if CPU usage is lower than usual.

I did think it was worth highlighting as this code will execute now for all source types (e.g. iislogs etc) which, given the context of the pull request, did not seem intentional. As for whether to follow up or not I don't have strong opinions on this - an acknowledgment is good enough.
